src/components/IconButton/src/index.tsx

```diff
@@ -35,6 +35,11 @@ export interface IconButtonProps extends BaseProps {
    * `small` is equivalent to the dense button styling.
    */
   size?: 'small' | 'medium';
+
+  /**
+   * Callback fired when the IconButton is clicked.
+   */
+  onClick?: (event: React.ChangeEvent<unknown>) => void;
```

Comment:
Shouldn't the type be a `React.MouseEvent` or something similar

Reply:
Yep you're right that's more precise.